internal/data/hash_storage.go

```diff
@@ -0,0 +1,100 @@
+package data
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/k8s"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+//go:generate mockgen -source=hash_storage.go -destination=mocks/mock_hash_storage.go
+
+type IHashStorage interface {
+	PrepareQuery(allHashData []*HashData, deploymentData *k8s.DeploymentData) (string, []any)
+	Get(dirPath string, deploymentData *k8s.DeploymentData) ([]*HashData, error)
+}
+
+type HashData struct {
+	ID           int
+	Hash         string
+	FullFileName string
+	Algorithm    string
+	NamePod      string
+}
+
+type HashStorage struct {
+	db     *sql.DB
+	alg    string
+	logger *logrus.Logger
+}
+
+// NewHashStorage creates a new HashStorage structure to work with the database table.
+func NewHashStorage(db *sql.DB, alg string, logger *logrus.Logger) *HashStorage {
+	return &HashStorage{
+		db:     db,
+		alg:    alg,
+		logger: logger,
+	}
+}
```

Comment:
We need one more function, I think.
When integrity check failed, we will restart the pod. And before this we need to delete info from the DB related to this pod in case where deployment has several pods and we still keep the deployment but want to restart single pod.